src/library_fs.js

```diff
@@ -1544,6 +1544,17 @@ mergeInto(LibraryManager.library, {
       var path = PATH.join2(typeof parent === 'string' ? parent : FS.getPath(parent), name);
       return FS.symlink(target, path);
     },
+    ensureFolder: function(path, canRead, canWrite) {
```

Comment:
I'll rename to `createFolders`, how is that? I prefer `Folders` to `DirectoryTree` as we already have `Folder` in similar methods.

I'll make it throw if blocked by a file, and add a comment explaining usage.